easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -96,6 +97,9 @@
 # prefix for names of local variables in easyconfig files
 LOCAL_VAR_PREFIX = 'local_'
 
+ARCH_HARD_FILTER = {
+    AARCH64: ['Yasm'],
```

Comment:
Hmm, I'm not a big fan of this approach...

I would rather implement something more flexible, where you can configure EasyBuild to opt-in to or opt-out of optional dependencies based on certain condition (i.e. `aarch64`, GPU system, etc.).

In this particular case, configuring EasyBuild with `--filter-deps=Yasm` should be sufficient?

Reply:
It's ugly for sure. Using `--filter-deps=Yasm` should work, but it would need to always be done so for aarch64.

Dynamic dependencies would be interesting, could spiral into "we depend on cuda if we see a compatible GPU", but absolute exceptions like Yasm would need to be defined somewhere in EasyBuild itself. I have no good/better ideas on how to achieve that.